gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -398,68 +398,74 @@ OGRFeature *OGRFlatGeobufLayer::GetFeature(GIntBig nFeatureId)
     return poFeature;
 }
 
-OGRErr OGRFlatGeobufLayer::querySpatialIndex() {
-    auto indexNodeSize = m_poHeader->index_node_size();
-    if (m_ignoreSpatialFilter || !m_poFilterGeom || m_queriedSpatialIndex || indexNodeSize == 0)
+OGRErr OGRFlatGeobufLayer::readIndex()
+{
+    if (m_queriedSpatialIndex)
         return OGRERR_NONE;
-    CPLDebug("FlatGeobuf", "Attempting spatial index query");
+
+    auto indexNodeSize = m_poHeader->index_node_size();
```

Comment:
I'd change to `auto const`-s where applicable. Otherwise it raises immediate questions to a reader: can these things change? when, in what conditions, and one has to read through the following implementation...